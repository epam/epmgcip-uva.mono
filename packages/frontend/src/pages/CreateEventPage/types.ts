import { Language, EventLanguageSpecificFields } from 'src/types';

export enum LanguageEvent {
  Toggle = 'toggle',
  Change = 'change',
  ClearAlert = 'clearAlert',
}

export interface LanguageReducerAction {
  event: LanguageEvent;
  language: Language;
  withApproval: boolean;
  update?: Partial<EventLanguageSpecificFields>;
}
