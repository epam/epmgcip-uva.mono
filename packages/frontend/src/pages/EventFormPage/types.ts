import { Language, EventLanguageSpecificFields } from 'src/types';

export enum LanguageEvent {
  Toggle = 'toggle',
  Change = 'change',
  ClearAlert = 'clearAlert',
  ReplaceData = 'replaceData',
}

export interface LanguageReducerAction {
  event: LanguageEvent;
  language: Language;
  withApproval: boolean;
  update?: Partial<EventLanguageSpecificFields>;
}
