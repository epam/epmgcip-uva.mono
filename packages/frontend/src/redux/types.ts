export const SET_USER_NAME = 'SET_USER_NAME';
export const SET_SHOW_MENU = 'SET_SHOW_MENU';