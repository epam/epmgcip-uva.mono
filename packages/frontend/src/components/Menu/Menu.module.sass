@use 'src/style/colors'
@import 'src/style/typography'

.menuWrapper
    z-index: -1
    position: absolute
    top: 0
    left: 0
    opacity: 0
    width: 100vw
    height: 100vh
    background-color: colors.$blue-dark
    padding: 8px 12px 0 30px
    box-sizing: border-box

    .menuContent
        max-width: calc(768px - 30px)
        display: flex
        flex-direction: column
        margin: 0 auto

    .menuCloseButton
        width: 30px

    .menuLink
        @extend .H3
        color: colors.$white
        padding-bottom: 24px

        &:hover
            text-decoration: underline

    .currentMenuLink
        text-decoration: underline

.modalCancelButton
    color: colors.$violet
    background-color: colors.$white
    border: 1px solid colors.$violet

    &:hover
        color: colors.$white
        background-color: colors.$blue-light
        border: none

.modalUserLogoutButton
    background-color: colors.$blue-light
    color: colors.$white

    &:hover
        background-color: colors.$blue-lightest