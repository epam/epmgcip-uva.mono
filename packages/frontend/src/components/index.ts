export { Header } from './Header/Header';
export { Button } from './Button/Button';
export { Menu } from './Menu/Menu';
export { Link } from './Link/Link';
export { Input } from './Input/Input';
export { Select } from './Select/Select';
export { Loader } from './Loader/Loader';
export { Notification } from './Notification/Notification';