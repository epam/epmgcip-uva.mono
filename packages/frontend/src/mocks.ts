import { IUser, UserRole, UserStatus } from "./types";

export const ADMIN_MOCK: IUser = {
    id: '00000000-0000-0000-0000-000000000000',
    name: 'Monroe Panela',
    telegramName: '@panela',
    role: UserRole.Admin,
    status: UserStatus.Active,
}

export const COORDINATOR_MOCK: IUser = {
    id: '00000000-0000-0000-0000-000000000001',
    name: 'John Doe',
    telegramName: '@john_doe',
    role: UserRole.Coordinator,
    status: UserStatus.Active,
}