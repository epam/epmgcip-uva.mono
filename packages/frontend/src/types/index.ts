import { EventStatus, IEvent, IUser, IVolunteer } from 'uva-shared';

export enum ScrollDirection {
  Up = 'up',
  Down = 'down',
}

export {
  CreateEventAlerts,
  EventStatus,
  Gender,
  ImageType,
  Language,
  Month,
  UserRole,
  UserStatus,
  languages,
  languagesShort,
} from 'uva-shared';

export type {
  EventLanguageSpecific,
  EventLanguageSpecificData,
  EventLanguageSpecificFields,
  IEvent,
  IUser,
  IVolunteer,
} from 'uva-shared';

export type FilterEventStatuses = EventStatus | 'all';

export enum UpdateEventAlerts {
  None = 10,
  ConfirmUpdateDraft,
  ConfirmUpdateActive,
  ConfirmPublishToTelegram,
  ConfirmDeleteFromTelegram,
  ShowEventPublishSuccess,
  ShowEventPublishCancellation,
  ConfirmAddNewLanguageSpecificData,
  ConfirmDefaultUpdate,
  ConfirmEventWithExistingStatusActive,
}

interface IData<T> {
  // first data load was done
  initialized: boolean;
  // error occured during data load
  error: boolean;
  // data is currently loading
  loading: boolean;
  // reached last page of data
  finished: boolean;
  // all loaded data
  data: T[];
}

export interface IState {
  editor: IUser;
  isMenu: boolean;
  usersList: IUser[];
  manageUsersPage: {
    userSearchInput: string;
    scrollSize: number;
    scrollDirection: ScrollDirection;
  };
  manageEventsPage: {
    statusFilter: FilterEventStatuses;
    scrollDirection: ScrollDirection;
    limit: number;
    scrollSize: number;

    data: IData<IEvent>;
  };
  loading: boolean;
  error: null | string;
}

export type IValidationError = { [key: string]: string };

export interface IOption {
  name: string;
  value: string;
}
