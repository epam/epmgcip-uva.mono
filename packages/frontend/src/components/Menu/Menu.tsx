import { Button } from 'src/components';
import css from './Menu.module.sass';
import CloseSvg from 'src/assets/cross.svg';
import { useDispatch } from 'react-redux';
import { Dispatch } from '@reduxjs/toolkit';
import { setMenu, setUserName } from 'src/redux/actions';
import { Link } from 'src/components';
import { useNavigate } from 'react-router-dom';
import {
  CREATE_EVENT_ROUTE,
  EVENTS_ROUTE,
  MANAGE_USERS_ROUTE,
  ROOT_ROUTE,
  VOLUNTEERS_ROUTE,
} from '../../constants';

const getMenuLinkClasses = (currentRoute: string, chekingRoute: string) =>
  currentRoute === chekingRoute
    ? `${css.menuLink} ${css.currentMenuLink}`
    : css.menuLink;

export const Menu = () => {
  const dispatch: Dispatch = useDispatch();
  const navigate = useNavigate();
  const currentRoute = window.location.pathname;

  const handleCloseMenu = () => {
    dispatch(setMenu(false));
  };
  const handleChangePage = (route: string) => {
    navigate(route);
    handleCloseMenu();
  };
  const handleLogout = () => {
    dispatch(setUserName(''));
    handleChangePage(ROOT_ROUTE);
  };

  return (
    <div id='menu' className={css.menuWrapper}>
      <div className={css.menuContent}>
        <Button onClick={handleCloseMenu}>
          <img className={css.menuCloseButton} src={CloseSvg} />
        </Button>
        <Link
          onClick={() => handleChangePage(EVENTS_ROUTE)}
          className={getMenuLinkClasses(currentRoute, EVENTS_ROUTE)}
        >
          Events
        </Link>
        <Link
          onClick={() => handleChangePage(CREATE_EVENT_ROUTE)}
          className={getMenuLinkClasses(currentRoute, CREATE_EVENT_ROUTE)}
        >
          Create Event
        </Link>
        <Link
          onClick={() => handleChangePage(VOLUNTEERS_ROUTE)}
          className={getMenuLinkClasses(currentRoute, VOLUNTEERS_ROUTE)}
        >
          Volunteers List
        </Link>
        <Link
          onClick={() => handleChangePage(MANAGE_USERS_ROUTE)}
          className={getMenuLinkClasses(currentRoute, MANAGE_USERS_ROUTE)}
        >
          Manage Stuff
        </Link>
        <Link onClick={handleLogout} className={css.menuLink}>
          Logout
        </Link>
      </div>
    </div>
  );
};
