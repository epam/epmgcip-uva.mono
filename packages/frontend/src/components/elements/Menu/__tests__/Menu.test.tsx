import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ADMIN_ACTIVE_MOCK } from 'src/mocks';
import { IAction, SET_EDITOR } from 'src/redux/types';
import { IUser } from 'src/types';
import { describe, expect, it, vi } from 'vitest';
import { Menu } from '../Menu';

const navigateMock = vi.fn();

const stateMock = {
  editor: {} as IUser,
  usersList: [] as IUser[],
  isMenu: false,
  loading: false,
  error: null,
};

const reducerMock = (state = stateMock, action: IAction) => {
  if (action.type === SET_EDITOR) {
    return {
      ...state,
      editor: action.payload,
    };
  }

  return state;
};

const storeMock = configureStore({ reducer: reducerMock });

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');

  return {
    ...actual,
    useNavigate: () => ({ navigate: navigateMock }),
  };
});

describe('Testing: Menu', () => {
  it('should render', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <Menu />
      </Provider>
    );
    storeMock.dispatch({ type: SET_EDITOR, payload: ADMIN_ACTIVE_MOCK });

    expect(container).toMatchSnapshot();
  });
});
