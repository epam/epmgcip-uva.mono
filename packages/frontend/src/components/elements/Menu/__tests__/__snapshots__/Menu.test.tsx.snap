// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: Menu > should render 1`] = `
<div>
  <div
    class="_menuWrapper_ec8375"
    id="menu"
  >
    <div
      class="_menuContent_ec8375"
    >
      <button
        class="_button_2842f2"
      >
        <img
          class="_menuCloseButton_ec8375"
          src="/packages/frontend/src/assets/cross.svg"
        />
      </button>
      <a
        class="_link_43716b _menuLink_ec8375"
      >
        События
      </a>
      <a
        class="_link_43716b _menuLink_ec8375"
      >
        Создать событие
      </a>
      <a
        class="_link_43716b _menuLink_ec8375"
      >
        Список волонтеров
      </a>
      <a
        class="_link_43716b _menuLink_ec8375"
      >
        Выйти
      </a>
    </div>
  </div>
</div>
`;
